### PR TITLE
ref(spans): Add more tags to span metrics

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Add`span.domain` and `span.status_code` tags to metrics.

#skip-changelog

### CODE REVIEW

File: relay-server/src/metrics_extraction/transactions/mod.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@ use std::collections::BTreeMap;
 
 use relay_common::{DurationUnit, EventType, MetricUnit, SpanStatus, UnixTimestamp};
 use relay_dynamic_config::{AcceptTransactionNames, TaggingRule, TransactionMetricsConfig};
+use relay_filter::csp::SchemeDomainPort;
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/transactions/mod.rs:5]: You could probably also use `url::Url` to reduce cross-dependencies.

File: relay-server/src/metrics_extraction/transactions/mod.rs [hunk 2]
```diff
@@ -472,6 +473,23 @@ fn extract_span_metrics(
...
                     span_tags.insert("span.action".to_owned(), act.to_owned());
                 }
+
+                if span_op == "http.client" {
+                    let domain = match span.description.value().and_then(|v| v.split_once(' ')) {
+                        Some((_method, url)) => {
+                            let url = SchemeDomainPort::from(url);
+                            match (url.domain, url.port) {
+                                (Some(dom), Some(port)) => Some(format!("{}:{}", dom, port)),
+                                (Some(dom), None) => Some(dom),
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/transactions/mod.rs]: nit: 
```suggestion
                                (Some(domain), Some(port)) => Some(format!("{}:{}", domain, port)),
                                (Some(domain), None) => Some(domain),
```

**Review by jjbayer — APPROVED** [review]
Would be great to have some unit tests for these.

File: relay-server/src/metrics_extraction/transactions/mod.rs (hunk 1 above)
Comment (iker-barriocanal) [reply relay-server/src/metrics_extraction/transactions/mod.rs:5]: TIL, thanks! I'll do that in a follow-up once we've rolled this out.

**Pull request merged**